src/make_watertight/Arc.cpp

```diff
@@ -0,0 +1,713 @@
+  // don't want the input edge
+  adj_edges.erase(edge);
+
+  // make sure the edge is oriented correctly
+  for (moab::Range::iterator i = adj_edges.begin(); i != adj_edges.end(); i++) {
+    const moab::EntityHandle* adj_end_verts;
+    result = MBI()->get_connectivity(*i, adj_end_verts, n_verts);
+    if (moab::MB_SUCCESS != result) {
+      MBI()->list_entity(*i);
+      std::cout << "result=" << result
+                << " could not get connectivity of edge" << std::endl;
+      return result;
+      //print_edge( *i );
+    }
+    assert(moab::MB_SUCCESS == result);
+    assert(2 == n_verts);
+    if (end_verts[1] != adj_end_verts[0])
+      i = adj_edges.erase(i) - 1;
+  }
+
+  /* The next edge could be ambiguous if more than one exists.This happens in
+     surfaces that are ~1D, and in surfaces that have pinch points
+     (mod13surf881). Although I didn't handle this case yet, if it occurs:
+     -Remember that a pinch point could have not just 2, but multiple ears.
+     -Select the next edge as the edge that shares the same triangle.
+     -This approach will only work if the pinch point also exists in the
+      geometric curves.
+     -If the pinch point does not exist in the geometric curves (~1D surfs),
+      there is no robust way to handle it. There should be an input assumption
+      that this never happens. "The faceting skin cannot have pinch points
+      unless they also occur in the surface's geometric curves."
+   */
+  if (0 == adj_edges.size()) {
+    next_edge = 0;
+  } else if (1 == adj_edges.size()) {
+    next_edge = adj_edges.front();
+  } else {
+    std::cout << "get_next_oriented_edge: " << adj_edges.size() <<
+              " possible edges indicates a pinch point." << std::endl;
+    result = MBI()->list_entity(end_verts[1]);
+    //assert(moab::MB_SUCCESS == result);
+    //return moab::MB_MULTIPLE_ENTITIES_FOUND;
+    next_edge = adj_edges.front();
+  }
+  return moab::MB_SUCCESS;
+}
+
+moab::ErrorCode Arc::create_loops_from_oriented_edges_fast(moab::Range edges,
+                                                           std::vector< std::vector<moab::EntityHandle> >& loops_of_edges,
+                                                           const bool debug) {
+  // place all edges in map
+  std::multimap<moab::EntityHandle, edge> my_edges;
+  moab::ErrorCode rval;
+  for (moab::Range::const_iterator i = edges.begin(); i != edges.end(); ++i) {
+    // get the endpoints of the edge
+    const moab::EntityHandle* endpts;
+    int n_verts;
+    rval = MBI()->get_connectivity(*i, endpts, n_verts);
+    if (moab::MB_SUCCESS != rval || 2 != n_verts) {
+      MB_CHK_SET_ERR(moab::MB_FAILURE, "could not get connectivity");
+    }
+    // store the edges
+    edge temp;
+    temp.edge = *i;
+    temp.v0   = endpts[0];
+    temp.v1   = endpts[1];
+    my_edges.insert(std::pair<moab::EntityHandle, edge>(temp.v0, temp));
+  }
+  std::cout << "error: function not complete" << std::endl;
```

Comment:
Bug: This method has an unreachable `return` call on line 280, blocked by the first `return moab::MB_FAILURE` here.  As written, this method always returns `moab::MB_FAILURE`.